enhancements/installer/connected-onprem-assisted-installer.md

```diff
@@ -0,0 +1,236 @@
+
+* the user's pull secret
+* a systemd unit file that runs all containers necessary to run the assisted installer
+
+Similar to downloading the current installer, the download will default to the
+latest available version of the assisted installer, RHCOS and OpenShift.
+
+### User Stories [optional]
+
+#### Story 1
+
+I want to install OpenShift while running the assisted installer and the entire
+installation workflow on infrastructure that I control. I do not want a third
+party cloud service to control the provisioning of my hosts from across the
+internet.
+
+#### Story 2
+
+I want to automate use of the assisted installer and expect reproducible
+results. Utilizing a cloud service does not provide sufficient control or
+guarantees of reproducibility.
+
+#### Story 3
+
+I want to test the assisted installer prior to production use and expect
+reproducible results. Utilizing a cloud service does not provide sufficient
+control or guarantees of reproducibility.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Single Use on Day 1 Only
+
+The on-premise assisted installer will run once per cluster. As a live ISO, it
+will not utilize any persistent storage. Upon completion of a cluster
+installation, the live ISO will be shut down and all state will vanish.
+
+#### Creating the Agent ISO
+
+The assisted installer generates a live ISO that can be booted on hosts and then
+used to provision them. It runs the Agent that communicates with the assisted
+installer service. The ISO needs to include:
+
+* a URL to where the assisted installer is running
+* a CA certificate so it can trust the assisted installer
+
+The assistant needs local access to a base RHCOS ISO in order to generate the
+Agent ISO. It will then run the following command, passing ignition either via
+`stdin` or as an additional argument.
+
+```
+coreos-installer iso embed -o /<output-path>/agent.iso /path/to/base/rhcos.iso
+```
+
+To facilitate local access, the base ISO image will be included inside the
+assisted installer's container image.
+
+#### S3
+
+The cloud implementation stores each generated ISO in an S3-like service. This
+enables the service to scale its storage requirements as it generates many ISOs
+for the creation of many clusters.
+
+The on-premise implementation will only be used once to create a single
+cluster, so it does not require a scalable storage solution. Instead of running
+an S3-like service in the on-premise implementation, the assisted installer
+service will write the generated ISO to its filesystem and serve that file
+through its API directly from the filesystem.
+
+### Risks and Mitigations
+
+Communication between the Agent and the assisted installer must be protected
+with TLS. The Agent must be able to trust the assisted installer service so
+that it does not receive rogue provisioning instructions.
+
+## Design Details
+
+### Open Questions [optional]
+
+#### RHCOS ISO image delivery
```

Comment:
Need to check if we can use the osmet embedded image included in 4.6 - https://github.com/coreos/coreos-installer/pull/197